File: LoginSHA256.py
import hashlib

# Registrasi User
print("=== REGISTRASI USER ===")
username = input("Masukkan Username : ")
password = input("Masukkan Password : ")

# Ubah password ke hash SHA-256
simpan_hash = hashlib.sha256(password.encode()).hexdigest()
simpan_user = username

# Tampilkan hasil hash
print("\nData berhasil disimpan!")
print("Username :", simpan_user)
print("Hash Password :", simpan_hash)

# Login User
print("\n=== LOGIN USER ===")
login_user = input("Masukkan Username : ")
login_pass = input("Masukkan Password : ")

# Verifikasi Login
login_hash = hashlib.sha256(login_pass.encode()).hexdigest()

if login_user == simpan_user and login_hash == simpan_hash:
    print("Status Login : BERHASIL")
else:
    print("Status Login : GAGAL")

File: LoginSHA256.java
import java.util.Scanner;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class LoginSHA256 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        String username, password;
        String simpanUser, simpanHash;

        // Registrasi User
        System.out.println("=== REGISTRASI USER ===");
        System.out.print("Masukkan Username : ");
        username = input.nextLine();

        System.out.print("Masukkan Password : ");
        password = input.nextLine();

        // Ubah password ke hash SHA-256
        simpanHash = hashSHA256(password);
        simpanUser = username;

        // Tampilkan hasil hash
        System.out.println("\nData berhasil disimpan!");
        System.out.println("Username : " + simpanUser);
        System.out.println("Hash Password : " + simpanHash);

        // Login
        System.out.println("\n=== LOGIN USER ===");
        System.out.print("Masukkan Username : ");
        String loginUser = input.nextLine();

        System.out.print("Masukkan Password : ");
        String loginPass = input.nextLine();

        // Verifikasi Login
        String loginHash = hashSHA256(loginPass);

        if (loginUser.equals(simpanUser) && loginHash.equals(simpanHash)) {
            System.out.println("Status Login : BERHASIL");
        } else {
            System.out.println("Status Login : GAGAL");
        }

        input.close();
    }

    // Method Hash SHA-256
    public static String hashSHA256(String text) {
        try {
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            byte[] hashBytes = md.digest(text.getBytes());

            StringBuilder sb = new StringBuilder();
            for (byte b : hashBytes) {
                sb.append(String.format("%02x", b));
            }

            return sb.toString();

        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }
}